File: android/app/src/main/kotlin/com/wrbug/gitbbs/flutterbridge/mmkv/method/SaveTokenCall.kt
package com.wrbug.gitbbs.flutterbridge.mmkv.method

import com.wrbug.gitbbs.flutterbridge.FlutterMethodCall
import com.wrbug.gitbbs.mmkv.UserKv
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel


class SaveTokenCall : FlutterMethodCall {
    override fun getMethodName() = "saveToken"

    override fun invoke(call: MethodCall, result: MethodChannel.Result) {
        if (call.hasArgument("token")) {
            UserKv.saveToken(call.argument<String>("token") ?: "")
            result.success(null)
            return
        }
        result.error("miss token", "", "")
    }
}

File: ios/Runner/FlutterBridge/mmkv/method/SaveTokenCall.swift
//
//  GetTokenCall.swift
//  Runner
//
//  Created by WrBug on 2019/3/7.
//  Copyright © 2019 The Chromium Authors. All rights reserved.
//

import Foundation
class SaveTokenCall: MethodCall {
    func getMethodName() -> String {
        return "saveToken"
    }
    
     func invoke(call: FlutterMethodCall, result: (Any?) -> Void) {
        var args =  call.arguments as! Dictionary<String, Any>
        UserKv.saveToken(args["token"]as! String)
        result(true)
    }
}
